clients/application-amqp/src/main/java/org/eclipse/hono/application/client/amqp/ProtonBasedCommandSender.java

```diff
@@ -109,43 +114,62 @@ public ProtonBasedCommandSender(
             final String contentType,
             final Buffer data,
             final String replyId,
-            final Map<String, Object> properties,
             final Duration timeout,
             final SpanContext context) {
 
-        return requestResponseClient
-                .sendCommand(tenantId, deviceId, command, contentType, data, replyId, properties, timeout, context);
+        return requestResponseClient.sendCommand(
+                tenantId,
+                deviceId,
+                command,
+                contentType,
+                data,
+                replyId,
+                timeout,
+                context);
     }
 
-    private Future<Void> sendCommand(final String tenantId, final String deviceId, final String command,
-            final String contentType, final Buffer data, final String correlationId, final String replyId,
-            final Map<String, Object> properties, final Span span) {
+    private Future<Void> sendCommand(
+            final String tenantId,
+            final String deviceId,
+            final String command,
+            final String contentType,
+            final Buffer data,
+            final String correlationId,
+            final String replyId,
+            final Span span) {
 
         return getOrCreateSenderLink(CommandConstants.NORTHBOUND_COMMAND_REQUEST_ENDPOINT, tenantId)
                 .recover(thr -> Future.failedFuture(StatusCodeMapper.toServerError(thr)))
                 .compose(sender -> {
-                    final String targetAddress = AddressHelper
-                            .getTargetAddress(CommandConstants.NORTHBOUND_COMMAND_REQUEST_ENDPOINT, tenantId, deviceId,
-                                    connection.getConfig());
```

Comment:
By not using `AddressHelper#getTargetAddress`, the address rewriting feature isn't used any more.
That made me wonder whether that should be the case.
But I think it's correct to not use the address rewriting since it's about the `to` address, not the link address here.

Reply:
The ["Address Rewriting" chapter](https://www.eclipse.org/hono/docs/admin-guide/hono-client-configuration/#address-rewriting) in the documentation doesn't explicitly mention the C&C endpoints, but a "multi-tenant messaging environment" using link addresses like `test-vhost/telemetry/DEFAULT_TENANT` would also want to use C&C link addresses with such a prefix, I guess.
As the application client is directly connected to the AMQP messaging system, it would also need to apply that address rewriting then.

Reply:
.. and for the *link* address, the `ProtonBasedCommandSender` does indeed use the address rewriting.
In any case, not rewriting the `to` address here is consistent with the fact that the Command Router can't currently deal with command messages that have a prefix like `test-vhost/` in the `to` address (no reverse rewriting done there).

Reply:
In general, I believe that we should remove the whole Address Rewriting functionality. My understanding is that it has been introduced to handle some special requirements of Red Hat's enMasse configuration/installation which I never understood in the first place. I also believe that nobody else has ever needed/used this feature. You could still use the standard AMQP 1.0 *open* frame's *hostname* property to share AMQP 1.0 based messaging infrastructure across multiple Hono installations. The Hono client also supports setting this property.

Reply:
> In any case, not rewriting the to address here is consistent with the fact that the Command Router can't currently deal with command messages that have a prefix like test-vhost/ in the to address (no reverse rewriting done there).

The C&C spec is also very clear about the format of the message *address* and does not allow for such a prefix.